### PR TITLE
[devicelab] update hostonly tests to use flutter directly or precache deps

### LINKED ISSUE #67370: Hostonly gradle tests rely on flutter doctor to download android dependencies
While removing the android gen_snapshot dependency from flutter doctor, the following tests failed:

* linux_gradle_plugin_bundle_test
* linux_gradle_plugin_fat_apk_test
* linux_module_host_with_custom_build_test
* mac_gradle_plugin_bundle_test
* mac_gradle_plugin_fat_apk_test
* mac_gradle_plugin_light_apk_test
* mac_module_host_with_custom_build_test
* win_gradle_plugin_bundle_test
* win_gradle_plugin_fat_apk_test
* win_gradle_plugin_light_apk_test
* win_module_host_with_custom_build_test

These tests should `flutter precache` any dependencies they need, and not rely on other commands happening to download the right binaries.

Additionally, the tool should probably check for dependencies in assemble.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes https://github.com/flutter/flutter/issues/67370

By running most of these executions through flutter, we get the benefit of the flutter error handling and precaching. IN the test where this is not feasible, call pre-cache directly.

### CODE REVIEW

File: dev/devicelab/bin/tasks/gradle_plugin_bundle_test.dart [hunk 1]
```diff
@@ -42,10 +45,6 @@ Future<void> main() async {
       });
 
       await runProjectTest((FlutterProject project) async {
-        if (Platform.isWindows) {
```
Comment (christopherfujino) [inline dev/devicelab/bin/tasks/gradle_plugin_bundle_test.dart:45]: We don't need this anymore?

**Review by christopherfujino — APPROVED** [review]
LGTM. This is a great improvement!

File: dev/devicelab/bin/tasks/gradle_plugin_bundle_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply dev/devicelab/bin/tasks/gradle_plugin_bundle_test.dart:45]: works on my computer (and on CI) 🤷🏻‍♂️

Comment (christopherfujino) [reply dev/devicelab/bin/tasks/gradle_plugin_bundle_test.dart:45]: ship it

**Pull request merged**